hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/EnrichmentProcessor.java

```diff
@@ -105,6 +107,12 @@ public void process(StructuredTrace trace) {
   }
 
   private void applyEnricher(Enricher enricher, StructuredTrace trace) {
+    Map<String, AttributeValue> attributeMap = trace.getAttributes().getAttributeMap();
+    AttributeValue dropTraceAttrValue = attributeMap.get(DROP_TRACE_ATTRIBUTE);
+    boolean shouldDropTrace = dropTraceAttrValue != null;
+    if (shouldDropTrace) {
```

Comment:
note: we should also drop the trace in enrichment processor. we can take care of it later